### DISCUSSION TITLE
Pre-task possibility

### DISCUSSION THREAD

**Discussion by redhandpl** [renovatebot/renovate#26593] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.117.1

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

Hi,
In documentation I see that there is a possibility to run some commands after upgrading modules and before commit (postUpgradeTasks).
What I would like to have is a possibility to run some pre-tasks which should be run after cloning the repository and before updating.
Is it possible?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
I think we need to define all possible times when custom tasks could/should be run and then adapt/extend postUpgradeTasks to be more generic. I can think of at least:

- postClone/preExtract
- preUpgrade
- postUpgrade

**Reply by redhandpl** [reply]
It would be even more awesome!

**Reply by viceice** [reply]
i think there's an open issue for pre upgrade tasks

**Reply by redhandpl** [reply, marked as answer]
Found it!
* https://github.com/renovatebot/renovate/issues/8804
* https://github.com/renovatebot/renovate/pull/20423

Thanks
